scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -897,139 +1068,199 @@ def test_as_euler_degenerate_compare_algorithms(seq_tuple, intrinsic):
+def test_inv(xp):
+    atol = 1e-12
     rnd = np.random.RandomState(0)
     n = 10
     # preserve use of old random_state during SPEC 7 transition
     p = Rotation.random(num=n, random_state=rnd)
+    p = Rotation.from_quat(xp.asarray(p.as_quat()))
     q = p.inv()
 
     p_mat = p.as_matrix()
     q_mat = q.as_matrix()
-    result1 = np.einsum('...ij,...jk->...ik', p_mat, q_mat)
-    result2 = np.einsum('...ij,...jk->...ik', q_mat, p_mat)
+    result1 = xp.asarray(np.einsum("...ij,...jk->...ik", p_mat, q_mat))
+    result2 = xp.asarray(np.einsum("...ij,...jk->...ik", q_mat, p_mat))
 
-    eye3d = np.empty((n, 3, 3))
-    eye3d[:] = np.eye(3)
+    eye3d = xp.empty((n, 3, 3))
+    eye3d = xpx.at(eye3d)[..., :3, :3].set(xp.eye(3))
 
-    assert_array_almost_equal(result1, eye3d)
-    assert_array_almost_equal(result2, eye3d)
+    xp_assert_close(result1, eye3d, atol=atol)
+    xp_assert_close(result2, eye3d, atol=atol)
 
 
-def test_inv_single_rotation():
+@pytest.mark.array_api_backends
+def test_inv_single_rotation(xp):
+    atol = 1e-12
     rng = np.random.default_rng(146972845698875399755764481408308808739)
     p = Rotation.random(rng=rng)
+    p = Rotation.from_quat(xp.asarray(p.as_quat()))
     q = p.inv()
 
     p_mat = p.as_matrix()
     q_mat = q.as_matrix()
     res1 = np.dot(p_mat, q_mat)
     res2 = np.dot(q_mat, p_mat)
 
-    eye = np.eye(3)
+    eye = xp.eye(3)
 
-    assert_array_almost_equal(res1, eye)
-    assert_array_almost_equal(res2, eye)
+    xp_assert_close(res1, eye, atol=atol)
+    xp_assert_close(res2, eye, atol=atol)
 
     x = Rotation.random(num=1, rng=rng)
+    x = Rotation.from_quat(xp.asarray(x.as_quat()))
     y = x.inv()
 
     x_matrix = x.as_matrix()
     y_matrix = y.as_matrix()
-    result1 = np.einsum('...ij,...jk->...ik', x_matrix, y_matrix)
-    result2 = np.einsum('...ij,...jk->...ik', y_matrix, x_matrix)
+    result1 = xp.linalg.matmul(x_matrix, y_matrix)
+    result2 = xp.linalg.matmul(y_matrix, x_matrix)
+
+    eye3d = xp.empty((1, 3, 3))
+    eye3d = xpx.at(eye3d)[..., :3, :3].set(xp.eye(3))
+
+    xp_assert_close(result1, eye3d, atol=atol)
+    xp_assert_close(result2, eye3d, atol=atol)
+
 
-    eye3d = np.empty((1, 3, 3))
-    eye3d[:] = np.eye(3)
+def test_inv_jax_compile():
+    jax = pytest.importorskip("jax")
 
-    assert_array_almost_equal(result1, eye3d)
-    assert_array_almost_equal(result2, eye3d)
+    r = Rotation.from_matrix(jax.numpy.eye(3))
+    inv = jax.jit(Rotation.inv)
+    jax.block_until_ready(inv(r))
 
 
-def test_identity_magnitude():
+@pytest.mark.array_api_backends
+def test_identity_magnitude(xp):
     n = 10
-    assert_allclose(Rotation.identity(n).magnitude(), 0)
-    assert_allclose(Rotation.identity(n).inv().magnitude(), 0)
+    r = Rotation.identity(n)
+    r = Rotation.from_quat(xp.asarray(r.as_quat()))
+    xp_assert_close(r.magnitude(), xp.asarray(0.0), check_shape=False)
+    xp_assert_close(r.inv().magnitude(), xp.asarray(0.0), check_shape=False)
```

Comment:
Why can't we test the output shape?

Reply:
The previous assert just checked against 0, so it also didn't cover the shape. Using `check_shape=False` replicates that as closely as possible. I have changed it now to include the shape using `xp.zeros(n)`.